### PR TITLE
W-13065850: add automatic module for gateway dependencies

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

**Comment by jcgrethe** [comment]
--validate

File: bom/pom.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
     <description>Mule runtime modules for BOMs</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.bom</javaModuleName>
```
Comment (elrodro83) [inline bom/pom.xml:18]: remove

File: bom/runtime-api/pom.xml [hunk 2]
```diff
@@ -15,6 +15,7 @@
     <description>Module that provides all the APIs versions provided by the Mule Runtime for a particular runtime version. Users of the APIs should use this BOM when depending on any Mule API.</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.bom.api</javaModuleName>
```
Comment (elrodro83) [inline bom/runtime-api/pom.xml:18]: remove

File: bom/runtime-impl/pom.xml [hunk 3]
```diff
@@ -14,6 +14,7 @@
     <description>A BOM project that provides all the implementation libraries required by the Mule runtime</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.bom.impl</javaModuleName>
```
Comment (elrodro83) [inline bom/runtime-impl/pom.xml:17]: remove

File: embedded/embedded-distribution/pom.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
     <description>Assembles the Mule core, modules, and transports into a single library (jar). This is useful for embedding Mule into an existing project.</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.embedded.distribution</javaModuleName>
```
Comment (elrodro83) [inline embedded/embedded-distribution/pom.xml:17]: remove

File: embedded/embedded-impl-bom/pom.xml [hunk 5]
```diff
@@ -15,6 +15,7 @@
     <description>This project contains all the specific dependencies of the embedded runtime implementation independent from the runtime version</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.embedded.bom.impl</javaModuleName>
```
Comment (elrodro83) [inline embedded/embedded-impl-bom/pom.xml:18]: remove

File: embedded/embedded-impl/pom.xml [hunk 6]
```diff
@@ -14,6 +14,7 @@
     <name>Mule Embedded Implementation</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.embedded.impl</javaModuleName>
```
Comment (elrodro83) [inline embedded/embedded-impl/pom.xml:17]: `org.mule.distribution.embedded.impl`

File: embedded/embedded-test/pom.xml [hunk 7]
```diff
@@ -13,6 +13,7 @@
     <name>Mule Embedded Implementation Tests</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.embedded.test</javaModuleName>
```
Comment (elrodro83) [inline embedded/embedded-test/pom.xml:16]: `org.mule.distribution.embedded.test`

File: embedded/pom.xml [hunk 8]
```diff
@@ -14,6 +14,7 @@
     <name>Mule Embedded</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.embedded</javaModuleName>
```
Comment (elrodro83) [inline embedded/pom.xml:17]: remove

File: mule-wrapper-additional-parameters-parser/pom.xml [hunk 9]
```diff
@@ -27,6 +27,7 @@
     </dependencies>
 
     <properties>
+        <javaModuleName>org.mule.runtime.wrapper.additional.parameters.parser</javaModuleName>
```
Comment (elrodro83) [inline mule-wrapper-additional-parameters-parser/pom.xml:30]: `org.mule.distribution.additional.parameters.parser`

File: pom.xml [hunk 10]
```diff
@@ -26,6 +26,7 @@
     </build>
 
 	<properties>
+        <javaModuleName>org.mule.runtime.distribution</javaModuleName>
```
Comment (elrodro83) [inline pom.xml:29]: remove

File: services-all/pom.xml [hunk 11]
```diff
@@ -18,6 +18,7 @@
     </description>
     
     <properties>
+        <javaModuleName>org.mule.runtime.service.all</javaModuleName>
```
Comment (elrodro83) [inline services-all/pom.xml:21]: remove

File: standalone-structure/pom.xml [hunk 12]
```diff
@@ -17,6 +17,7 @@
     <description>Defines the basic structure of a Mule standalone distribution in the filesystem.</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.standalone.structure</javaModuleName>
```
Comment (elrodro83) [inline standalone-structure/pom.xml:20]: remove

File: standalone/pom.xml [hunk 13]
```diff
@@ -17,6 +17,7 @@
     <description>Includes the Mule server with all of its APIs and internal modules.</description>
 
     <properties>
+        <javaModuleName>org.mule.runtime.standalone</javaModuleName>
```
Comment (elrodro83) [inline standalone/pom.xml:20]: remove

File: tests/pom.xml [hunk 14]
```diff
@@ -13,6 +13,7 @@
     <name>Mule Distribution Tests</name>
 
     <properties>
+        <javaModuleName>org.mule.runtime.distribution.tests</javaModuleName>
```
Comment (elrodro83) [inline tests/pom.xml:16]: `org.mule.distribution.test`

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — DISMISSED** [review]

**Comment by jcgrethe** [comment]
--validate

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**